pkg/didcomm/crypto/authcrypt/encrypt_test.go

```diff
@@ -0,0 +1,103 @@
+/*
+Copyright SecureKey Technologies Inc. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package authcrypt
+
+import (
+	"bytes"
+	"encoding/json"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"golang.org/x/crypto/nacl/box"
+)
+
+func TestEncrypt(t *testing.T) {
+	var err error
+	// create temporary keys for testing
+	sendEcKey := keyPair{}
+	sendEcKey.pub, sendEcKey.priv, err = box.GenerateKey(RandReader)
+	require.NoError(t, err)
+
+	recipient1Key := keyPair{}
+	recipient1Key.pub, recipient1Key.priv, err = box.GenerateKey(RandReader)
+	require.NoError(t, err)
+
+	recipient2Key := keyPair{}
+	recipient2Key.pub, recipient2Key.priv, err = box.GenerateKey(RandReader)
+	require.NoError(t, err)
+
+	recipient3Key := keyPair{}
+	recipient3Key.pub, recipient3Key.priv, err = box.GenerateKey(RandReader)
+	require.NoError(t, err)
+
+	badKey := keyPair{
+		pub:  nil,
+		priv: nil,
+	}
+	t.Run("Error test case: Create a new AuthCrypter with bad encryption algorithm", func(t *testing.T) {
+		_, e := NewAuthCrypter(sendEcKey, []keyPair{recipient1Key, recipient2Key, recipient3Key}, "BAD")
+		require.Error(t, e)
+	})
+
+	t.Run("Error test case: Create a new AuthCrypter with bad sender key", func(t *testing.T) {
+		_, e := NewAuthCrypter(badKey, []keyPair{recipient1Key, recipient2Key, recipient3Key}, "XC20P")
+		require.Error(t, e)
+	})
+
+	t.Run("Error test case: Create a new AuthCrypter with bad recipient key", func(t *testing.T) {
+		_, e := NewAuthCrypter(sendEcKey, []keyPair{badKey}, "XC20P")
+		require.Error(t, e)
+	})
+
+	t.Run("Success test case: Create a valid AuthCrypter for XChachaPoly1035 encryption (alg: XC20P)", func(t *testing.T) {
+		crypter, e := NewAuthCrypter(sendEcKey, []keyPair{recipient1Key, recipient2Key, recipient3Key}, "XC20P")
+		require.NoError(t, e)
+		require.NotEmpty(t, crypter)
+		enc, e := crypter.Encrypt("lorem ipsum dolor sit amet")
+		require.NoError(t, e)
+		require.NotEmpty(t, enc)
+
+		m, e := prettyPrint(enc)
+		require.NoError(t, e)
+		t.Logf("Encryption with XC20P: %s", m)
+	})
+
+	t.Run("Success test case: Create a valid AuthCrypter for ChachaPoly1035 encryption (alg: C20P)", func(t *testing.T) {
+		crypter, e := NewAuthCrypter(sendEcKey, []keyPair{recipient1Key, recipient2Key, recipient3Key}, "C20P")
+		require.NoError(t, e)
+		require.NotEmpty(t, crypter)
+		enc, e := crypter.Encrypt("lorem ipsum dolor sit amet")
+		require.NoError(t, e)
+		require.NotEmpty(t, enc)
+
+		m, e := prettyPrint(enc)
+		require.NoError(t, e)
+		t.Logf("Encryption with C20P: %s", m)
+
+		t.Run("Error test Case: use a valid AuthCrypter but scramble the nonce size", func(t *testing.T) {
```

Comment:
This test case should be a sibling of the parent

Reply:
it's re-using the crypter

Reply:
I've unexpected all crypter's fields, so nonce testing is internal and should not be exposed outside the library. It should be fine to re-use the crypter for 'internally' testing the nonceSize value.